test/test_jit.py

```diff
@@ -3019,6 +3019,25 @@ def stuff3(x):
             return torch.ones(x), x
         self.checkScript(stuff3, ([3, 2],))
 
+    def test_for_in_tensors(self):
```

Comment:
Add a test for iterating a zero-dim tensor as well, should say something like `TypeError: iteration over a 0-d tensor`

Reply:
@driazati 
Added two more tests to 0 test-dim and scalar iterables. 0-dim shouldn't throw either in python or pytorch jit. The scalar testcase does throw, but the error message is a bit misleading. There's a bunch of test cases that target this particular error message. I'd suggest we fix those in a separate PR as it doesn't have a lot to do with for-in for tensor targets.

Reply:
The tests still all have at least 1-dim, `torch.tensor(scalar)` makes a 0-dim tensor which it doesn't really make sense to iterate over, see:

```python
>>> torch.rand(0).size()
torch.Size([0])
>>> torch.tensor(2).size()
torch.Size([])
>>> for i in torch.tensor(2):
...     print(i)
... 
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
    raise TypeError('iteration over a 0-d tensor')
TypeError: iteration over a 0-d tensor
```